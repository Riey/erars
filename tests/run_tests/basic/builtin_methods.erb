@SYSTEM_TITLE

ADDDEFCHARA
TARGET = 0

PRINTFORML MAX {MAX(6)} {MAX(0, 100, 1999, -23423)}
PRINTFORML MIN {MIN(5)} {MIN(0, 100, 1999, -23423)}
PRINTFORML LIMIT {LIMIT(-1, 0, 100)} {LIMIT(35, 0, 100)} {LIMIT(10423, 0, 100)}
PRINTFORML GROUPMATCH {GROUPMATCH(3, 1, 1, 2, 3, 3, 3, 1421)} {GROUPMATCH("123", "1", "2", "#", "123", "123", "", "2354123")}

LOCAL = 1
LOCAL:1 = 2
ABL:0 = 1
ABL:3 = 2
ABL:5 = 3
PRINTFORML SUMARRAY {SUMARRAY(LOCAL:1, 0, 2)} {SUMARRAY(ABL:0:0, 0, 6)}

LOCALS = 레이무
STRLENSU LOCALS
SUBSTRINGU LOCALS, RESULT - 1, 1
PRINTFORML STRCOUNT {STRCOUNT("142333", "42")} {STRCOUNT("레레 레이무", "레")}
PRINTFORML SUBSTRINGU %RESULTS%
PRINTFORML REPLACE {REPLACE("1111", "11", "223")}

PRINTFORML GETBIT {GETBIT(0b1000, 3)} {GETBIT(0b1000, 2)}

VARSET LOCALS
LOCALS:0 = 1
LOCALS:1 = 2
LOCALS:2 = 3
LOCALS:3 = 4
LOCALS:4 = 5

PRINTFORML STRJOIN %STRJOIN(LOCALS, ",", 0, 5)% %STRJOIN(LOCALS, "|", 1, 3)%
