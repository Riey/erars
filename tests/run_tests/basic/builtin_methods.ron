[
    Print("MAX 6 1999"), NewLine,
    Print("MIN 5 -23423"), NewLine,
    Print("LIMIT 0 35 100"), NewLine,
    Print("GROUPMATCH 3 2"), NewLine,
    Print("SUMARRAY 3 6"), NewLine,
    Print("STRCOUNT 1 3"), NewLine,
    Print("SUBSTRINGU ë¬´"), NewLine,
    Print("GETBIT 1 0"), NewLine,
]