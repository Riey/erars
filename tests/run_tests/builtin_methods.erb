@SYSTEM_TITLE
PRINTFORML LIMIT {LIMIT(-1, 0, 100)} {LIMIT(35, 0, 100)} {LIMIT(10423, 0, 100)}
