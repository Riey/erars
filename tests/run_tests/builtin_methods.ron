[Print("LIMIT 0 35 100"), NewLine]
