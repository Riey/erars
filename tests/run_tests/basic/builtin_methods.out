MAX 6 1999
MIN 5 -23423
LIMIT 0 35 100
POWER 8 1 0
GROUPMATCH 3 2
SUMARRAY 3 6
SUMCARRAY 4 4
STRCOUNT 1 3
SUBSTRINGU 무
REPLACE 223223
GETBIT 1 0
STRJOIN 1,2,3,4,5 2|3|4
FINDELEMENT 0 FINDLASTELEMENT 4
FINDELEMENT tw bug 1
TOUPPER ABC TOLOWER abc TOHALF ABC TOFULL ＡＢＣ
ISNUMERIC 1
ENCODETOUNI
3495051
0000
51
GETCONFIG 19 D2Coding
EXISTCSV 0 1
VARSIZE 1000 1000
